contracts/utils/CREATE2Factory.sol

```diff
@@ -0,0 +1,64 @@
+pragma solidity ^0.5.0;
+
+/**
+ * @title CREATE2Factory
+ *
+ * @dev This contract factory focus on the use of CREATE2 EVM opcode for
+ * contracts deployment. It also provides a function to precompute the address
+ * where the smart contracts with the specified salt and bytecode would be
+ * deployed.
+ */
+contract CREATE2Factory {
```

Comment:
I think we could simply call this `Create2`, and we'd end up with functions like `Create2.deploy`, etc. Not a fan of exposing such a low level detail in the name, but it's succinct, and everyone already knows about it. I'm open to alternatives.

Reply:
Couldn't agree more. :+1: